### PR TITLE
feat: add Expo React Native docs

### PR DESCRIPTION
**Pull request by iykazrji**
# Pull Request Checklist

- [ ] Did you add new tests and confirm existing tests pass? (`yarn test`)
- [x] Did you update relevant docs? (docs are found in the `site` folder, and guidelines for updating/adding docs can be found in the [contribution guide](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md))
- [x] Do your commits follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [x] Does your PR title also follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [x] If you have a breaking change, is it [correctly reflected in your commit message](https://www.conventionalcommits.org/en/v1.0.0/#examples)? (e.g. `feat!: breaking change`)
- [x] Did you run lint (`yarn lint:check`) and fix any issues? (`yarn lint:write`)
- [x] Did you follow the [contribution guidelines](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md)?

---

## PR-Codex overview
This PR focuses on reorganizing and enhancing the sidebar navigation for React Native resources, integrating new content related to the `react-native-signer` and `getting-started` sections, and updating references across various files.

### Detailed summary
- Deleted the `react-native.mdx` file.
- Added `react-native-signer.ts` and `getting-started.ts` sidebar items.
- Updated `index.ts` to include new sidebar items.
- Modified sidebar references in multiple files to incorporate new structure.
- Enhanced content in `overview.mdx`, `setup-guide.mdx`, and `authenticating-users.mdx` for clarity and guidance.
- Updated links in `faqs.mdx` to point to new resources.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: site/pages/react-native/getting-started.mdx [hunk 1]
```diff
@@ -1,16 +1,3 @@
----
-title: Using within React Native applications
-description: A guide on integrating Account Kit within a React Native application
----
```
Comment (moldy530) [inline site/pages/react-native/getting-started.mdx:4]: We still need a header like this up here for the page

File: site/pages/react-native/getting-started.mdx [hunk 2]
```diff
@@ -1,16 +1,3 @@
...
-
-# Using within React Native applications
-
-This hasn't been tested extensively, but it is possible to use Account Kit within a React Native application. We've built a simple example using expo [here](https://github.com/alchemyplatform/aa-sdk-rn-expo/tree/main). This guide assumes you're using Expo, but the same principles should apply to a bare React Native app as well.
-
-:::warning
-The Alchemy Signer is not yet supported in React Native applications. For 3rd party signer support in React Native, refer to your signer provider's documentation.
-:::
```
Comment (moldy530) [inline site/pages/react-native/getting-started.mdx:12]: we should update this copy to call out that RN signer support is in alpha

File: site/pages/react-native/overview.mdx [hunk 3]
```diff
@@ -0,0 +1,12 @@
+---
+title: Using within React Native applications
+description: A guide on integrating Account Kit within a React Native application
+---
+
+# Using within React Native applications
+
+This hasn't been tested extensively, but it is possible to use Account Kit within a React Native application. We've built a simple example using expo [here](https://github.com/alchemyplatform/aa-sdk-rn-expo/tree/main). This guide assumes you're using Expo, but the same principles should apply to a bare React Native app as well.
```
Comment (moldy530) [inline site/pages/react-native/overview.mdx:8]:
```suggestion
We've built a simple example using expo [here](https://github.com/alchemyplatform/aa-sdk-rn-expo/tree/main). This guide assumes you're using Expo, but the same principles should apply to a bare React Native app as well.
```

File: site/pages/react-native/getting-started.mdx
Comment (moldy530) [inline site/pages/react-native/getting-started.mdx:1]: the bottom of this file still includes outdated references to package names

File: site/pages/react-native/signer/expo-installation.mdx
Comment (moldy530) [inline site/pages/react-native/signer/expo-installation.mdx:1]: we should call this `expo-quickstart` and include some sample code at the bottom that shows how you init the signer and maybe login with it

Comment (moldy530) [reply site/pages/react-native/signer/expo-installation.mdx:1]: ignore me! you've got it in a different PR

File: site/pages/react-native/getting-started.mdx (hunk 1 above)
Comment (moldy530) [inline site/pages/react-native/getting-started.mdx:4]: We still need this

File: site/pages/react-native/getting-started.mdx [hunk 4]
```diff
@@ -1,16 +1,3 @@
...
-
 ## Getting Started
 
 ### Upgrade to the latest Beta version of Expo
```
Comment (moldy530) [inline site/pages/react-native/getting-started.mdx:3]: Is this still required? How much of this page actually still needs to exist? It was primarily focused on getting started with expo

File: site/pages/react-native/signer/expo-installation.mdx [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+:::
+
+## Add Supporting Dependencies
+
+To ensure the signer works properly, you'll need to add the following dependencies to your project:
+
+:::code-group
+
+```bash [npm]
+npm install @account-kit/signer react-native-mmkv zustand abitype
```
Comment (moldy530) [inline site/pages/react-native/signer/expo-installation.mdx:44]: woah... I wonder why RN doesn't pull in these deps (specifically the account-kit/signer and zustand and abitype ones)

File: site/pages/react-native/getting-started.mdx (hunk 4 above)
Comment (iykazrji) [reply site/pages/react-native/getting-started.mdx:3]: Yeah I think it should, the rest of the guides go over specifictly the Signer, but for setting up the rest of `account-kit` having a guide would be quite useful IMO.

File: site/pages/react-native/signer/expo-installation.mdx (hunk 5 above)
Comment (iykazrji) [reply site/pages/react-native/signer/expo-installation.mdx:44]: Yeah it's quite odd tbh, warrants a deeper look into how metro loads up dependencies of dependencies.

**Comment by iykazrji** [comment]
Closing this in favor of https://github.com/alchemyplatform/aa-sdk/pull/1181

**Pull request closed without merging**